### ISSUE TITLE
Feature request: add validation that rich text fields don’t contain empty headings

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#6526]
### Is your proposal related to a problem?

Prompted by a discussion on https://github.com/wagtail/wagtail/pull/6272 / https://github.com/wagtail/wagtail/issues/6090. It’s problematic for screen reader users if there are empty headings on pages. There is no validation for this in rich text fields.

### Describe the solution you'd like

In StreamField it’s very easy for developers to make sure heading blocks are required (using [`CharBlock`](https://docs.wagtail.io/en/stable/topics/streamfield.html?highlight=charblock#charblock), required is the default). We should have something equivalent for rich text.

The default should be for all of h1, h2, h3, h4, h5, h6 to require having text in the block if present. It would be nice if this was configurable for people who would want paragraphs (#4431) or list items to be required, but this doesn’t have as clear of an impact accessibility-wise so shouldn’t be the default.

### Describe alternatives you've considered

This could be implemented per-project, but really since Wagtail provides rich text out of the box, this type of validation should also be there out of the box. There are no use cases we would want to support for empty headings.

### Additional context

I think this is a stretch but it would be nice if this was:

- Extensible to support other types of rich text validation in the future (for example, validating that link text isn’t "click here")
- Configurable by implementers of Wagtail (for example, to make their own custom blocks required, or to restrict link text further)

**Comment by thibaudcolas** [comment]
Discussing this with @helenb today, and in the accessibility team’s meeting last week, a simpler implementation might just be to strip those heading levels from rich text on save.

Personally I’m more inclined to use validation so end users are more aware of the issue, but happy with the more radical approach as well if there is agreement it’s a good fit for Wagtail (and whether it should be opt-out, opt-in, configurable in some way).

**Comment by Scotchester** [comment]
Thinking about this sort of validation today, as we've seen some unfortunate behaviors recently that we'd love to prevent:

1. Making sure heading levels are not skipped (when available as features in a Draftail field)
2. Making sure that entire paragraphs in a rich text field/block are not bolded/italicized/capitalized as a stand-in for headings (when headings are not available in Draftail and we are trying to force them to be in their own block)

**Comment by thibaudcolas** [comment]
As of #8805, we’re making empty heading blocks much more visually prominent. This strikes me as an appropriate solution to this issue, even if there is no actual validation, so I’ll close this now.
